geozero/src/csv/csv_writer.rs

```diff
@@ -0,0 +1,484 @@
+        self.wkt_writer
+            .circularstring_end(self.offset_geom_idx(idx))
+    }
+    fn compoundcurve_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.wkt_writer.compoundcurve_begin(size, idx)
+    }
+    fn compoundcurve_end(&mut self, idx: usize) -> Result<()> {
+        self.wkt_writer.compoundcurve_end(self.offset_geom_idx(idx))
+    }
+    fn curvepolygon_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.wkt_writer.curvepolygon_begin(size, idx)
+    }
+    fn curvepolygon_end(&mut self, idx: usize) -> Result<()> {
+        self.wkt_writer.curvepolygon_end(self.offset_geom_idx(idx))
+    }
+    fn multicurve_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.wkt_writer.multicurve_begin(size, idx)
+    }
+    fn multicurve_end(&mut self, idx: usize) -> Result<()> {
+        self.wkt_writer.multicurve_end(self.offset_geom_idx(idx))
+    }
+    fn multisurface_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.wkt_writer.multisurface_begin(size, idx)
+    }
+    fn multisurface_end(&mut self, idx: usize) -> Result<()> {
+        self.wkt_writer.multisurface_end(self.offset_geom_idx(idx))
+    }
+    fn triangle_begin(&mut self, tagged: bool, size: usize, idx: usize) -> Result<()> {
+        self.wkt_writer.triangle_begin(tagged, size, idx)
+    }
+    fn triangle_end(&mut self, tagged: bool, idx: usize) -> Result<()> {
+        self.wkt_writer.triangle_end(tagged, idx)
+    }
+    fn polyhedralsurface_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.wkt_writer.polyhedralsurface_begin(size, idx)
+    }
+    fn polyhedralsurface_end(&mut self, idx: usize) -> Result<()> {
+        self.wkt_writer
+            .polyhedralsurface_end(self.offset_geom_idx(idx))
+    }
+    fn tin_begin(&mut self, size: usize, idx: usize) -> Result<()> {
+        self.wkt_writer.tin_begin(size, idx)
+    }
+    fn tin_end(&mut self, idx: usize) -> Result<()> {
+        self.wkt_writer.tin_end(self.offset_geom_idx(idx))
+    }
+}
+
+// Writing Wkt is delegated to the WktWriter, and it's output buffered to Vec, to be eventually
+// incorporated in the output of the CSV writer. Is there a better way?
+mod buffering_wkt_writer {
+    use crate::error::Result;
+    use crate::{wkt::WktWriter, CoordDimensions, GeomProcessor};
+
+    #[derive(Default)]
+    pub(crate) struct BufferingWktWriter {
+        buffer: Vec<u8>,
+        pub(crate) dims: CoordDimensions,
+    }
+
+    impl BufferingWktWriter {
+        pub(crate) fn new() -> Self {
+            Self::default()
+        }
+
+        pub(crate) fn clear(&mut self) {
+            self.buffer.clear()
+        }
+
+        pub(crate) fn bytes(&self) -> &[u8] {
+            &self.buffer
+        }
+
+        fn wkt_writer(&mut self) -> WktWriter<'_, Vec<u8>> {
+            WktWriter::new(&mut self.buffer)
+        }
+    }
+
+    impl GeomProcessor for BufferingWktWriter {
```

Comment:
All this explicit delegation is kind of tedious. But I'm not sure of a better way to embed the output of WktWriter into the CSVWriter.

Reply:
I hope these delegations won't be necessary with the event API anymore.